benches/extract.rs

```diff
@@ -42,25 +85,27 @@ fn extract_basic(bench: &mut Bencher) {
             let outdir = TempDir::new_in(parent.path(), "bench-subdir")
                 .unwrap()
                 .into_path();
-            readable_archive.extract(outdir).unwrap();
+            archive.extract(outdir).unwrap();
         });
     });
 }
 
-#[cfg(all(feature = "parallelism", feature = "bzip2", unix))]
-const DECOMPRESSION_THREADS: usize = 8;
+fn extract_basic_static(bench: &mut Bencher) {
+    let mut archive = STATIC_TEST_ARCHIVE.lock().unwrap();
+    do_extract_basic(bench, &mut archive);
+}
+
+fn extract_basic_dynamic(bench: &mut Bencher) {
+    let mut archive = DYNAMIC_TEST_ARCHIVE.lock().unwrap();
+    do_extract_basic(bench, &mut archive);
+}
 
-#[cfg(all(feature = "parallelism", feature = "bzip2", unix))]
-fn extract_split(bench: &mut Bencher) {
-    let readable_archive = get_test_archive().unwrap();
-    let total_size: u64 = readable_archive
-        .decompressed_size()
-        .unwrap()
-        .try_into()
-        .unwrap();
+#[cfg(all(feature = "parallelism", unix))]
+fn do_extract_split(bench: &mut Bencher, archive: &ZipArchive<fs::File>) {
+    let total_size: u64 = archive.decompressed_size().unwrap().try_into().unwrap();
 
     let params = ExtractionParameters {
-        decompression_threads: DECOMPRESSION_THREADS,
+        decompression_threads: num_cpus::get() / 3,
```

Comment:
What will the other 2/3 of the CPUs be doing? Also, does this need to be clamped to at least 1?